File: examples/bakery-interaction-examples/make-payment-and-ship-items/src/main/scala/webshop/webservice/ShipItems.scala
package webshop.webservice

import com.ing.baker.recipe.annotations.FiresEvent

import scala.concurrent.Future

case class Item(itemId: String)

case class ShippingConfirmed()

case class ShippingAddress(address: String)

case class ReservedItems(items: List[Item], data: Array[Byte])

trait ShipItems {

  @FiresEvent(oneOf = Array(classOf[ShippingConfirmed]))
  def apply(shippingAddress: ShippingAddress, reservedItems: ReservedItems): Future[ShippingConfirmed]
}

File: examples/baker-example/src/main/java/webshop/simple/ShipItems.java
package webshop.simple;

import com.ing.baker.recipe.annotations.FiresEvent;
import com.ing.baker.recipe.annotations.RequiresIngredient;
import com.ing.baker.recipe.javadsl.Interaction;
import webshop.simple.ingredients.ReservedItems;
import webshop.simple.ingredients.ShippingAddress;

public interface ShipItems extends Interaction {

    class ShippingConfirmed {}

    @FiresEvent(oneOf = {ShippingConfirmed.class})
    ShippingConfirmed apply(
            @RequiresIngredient("shippingAddress") ShippingAddress shippingAddress,
            @RequiresIngredient("reservedItems") ReservedItems reservedItems
    );
}
